YALB-722: Interactive grid

ModulesUnraveled opened the pull request:
## [YALB-722: Interactive grid](https://yaleits.atlassian.net/browse/YALB-722)

### Description of work
- Creates the interactive Grid component and functionality

### Testing Link(s)
- [ ] Navigate to the [Interactive Grid Story](https://deploy-preview-153--dev-component-library-twig.netlify.app/?path=/story/organisms-galleries--interactive-grid)

### Functional Review Steps
- [x] Verify you can install the component with the CLI
- [ ] Check keyboard navigation (arrow keys, and tab nav) and clicking the controls
- [ ] Also check swipe on mobile, and "click and drag" on desktop

### Design Review
- [ ] Verify the designs match the [Figma Designs](https://www.figma.com/file/OVfmCW5s1gI7xnrM8ibX2y/UI-Kit-Alpha-%5BYaleSites%5D)
- [x] Review any new [Percy builds](https://percy.io/95144ff9/component-library-twig)

### Accessibility Review
- [ ] Verify the component meets Accessibility requirements
- [ ] Check keyboard navigation (arrow keys, and tab nav) and clicking the controls
- [ ] Also check swipe on mobile, and "click and drag" on desktop

ahitson commented:
- Thinking the close button and arrow navs should adjust to the width of the image instead of hanging so far outside. 
- Use spacing 1 between the numbers.
- Slightly reduce the size of the interactive cue (lil arrow circle) when it's expanded on hover.
- Can the cue also be responsive so it takes up less space when the image gets smaller?
- Sometimes the numbers get a scroll bar, too. (attached)


<img width="584" alt="Screen Shot 2022-10-11 at 5 27 33 PM" src="https://user-images.githubusercontent.com/80782149/195210028-854e9f55-70be-47d7-9bbb-bb3a85e7fb8e.png">

miketullo95 commented:
There is an alignment issue with the header compared to the content shown below

![Screen Shot 2022-10-12 at 8 59 08 AM](https://user-images.githubusercontent.com/98550395/195349012-732e4d6b-8996-4655-851f-08e6ea62bb15.png)

ModulesUnraveled commented:
@ahitson @miketullo95 I think I've addressed all of the issues you brought up. Please re-review, and let me know if there's anything else!

codeb-a reviewed (approved):
Can verify install with the CLI.

ntay90 commented:
@ModulesUnraveled The buttons will need to be labeled for screen reader users: each image button should probably say that it's going to pop out, and the buttons within the carousel modal that opens all need labels as well.

Additionally, keyboard users are unable to scroll the content that is truncated.

- [] Label image and carousel buttons.
- [] Make the truncated content focusable for keyboard users to scroll.

<img width="1042" alt="Screen Shot 2022-10-13 at 8 53 18 AM" src="https://user-images.githubusercontent.com/69398223/195616414-e78d1bb6-a6d3-4029-8cc3-d7937971676b.png">
<img width="937" alt="Screen Shot 2022-10-13 at 8 56 17 AM" src="https://user-images.githubusercontent.com/69398223/195616973-512bf9cf-ab35-49db-9e28-7672c693b97c.png">

ModulesUnraveled commented:
@ntay90 I'm not sure what the right way to do this is:
> each image button should probably say that it's going to pop out

I think I've addressed all of your other requests. If you can point me to some docs or something for that last one, I can make it happen.

ntay90 commented:
@ModulesUnraveled Is it possible to add some visually hidden text following the image saying it's going to pop out?

ModulesUnraveled commented:
@ntay90 Yes, we can do that. I couldn't find any recommendations on the verbiage online, so I just went with "Open this image in a modal". If you have another suggestion, let me know. Otherwise, it's ready for review with that text.

ModulesUnraveled commented:
@ntay90 I added the `aria-current` to the `button` element. I thought that would make the most sense, but if it should be somewhere else, let me know.

I also added the visually hidden text. Let me know if that works!

ntay90 reviewed (approved):
@ModulesUnraveled That works!

callinmullaney reviewed (changes requested):
This is looking REEEALLLY close. Couple things I noticed so far:
1. In Firefox the icon for expanding isn't centered: 
![Screenshot from 2022-10-19 16-30-06](https://user-images.githubusercontent.com/369018/196809014-af31ac4d-7477-479e-9ec0-0d0921fcd79d.png)
2. Not directly an issue with the interactive grid but the Breadcrumbs module is throwing a JS error which then prevents the grid from functioning as expected.
![Screenshot from 2022-10-19 16-36-13](https://user-images.githubusercontent.com/369018/196809374-2ee75b9f-1459-44e2-90a7-fddaed51d262.png)

callinmullaney commented on components/03-organisms/galleries/media-grid/_yds-media-grid-modal-item--content.twig:
```diff
@@ -0,0 +1,25 @@
+{% set modal_item__heading = modal_item__heading|default(item.media_grid__modal__heading) %}
```
This file is missing Available Variables comments

callinmullaney commented on components/03-organisms/galleries/media-grid/_yds-media-grid-modal.scss:
```diff
@@ -1,27 +1,236 @@
+  // Stylelint is mis-interpreting the `/` below as a division operator, rather
+  // than as a part of the grid-template sizing stuff. So, we need to disable
+  // that check for this chunk of code.
+  /* stylelint-disable scss/operator-no-newline-before */
```
This seems fine to disable stylint here. The way you have `grid-template` formatted here is much easier to read, :tada:

ModulesUnraveled commented:
@callinmullaney I've addressed all of your concerns. Please take another look!

callinmullaney reviewed (approved):
:clap: :shipit:
